mlir/lib/Transforms/RemoveDeadValues.cpp

```diff
@@ -742,7 +742,49 @@ static void processBranchOp(BranchOpInterface branchOp, RunLivenessAnalysis &la,
 static void cleanUpDeadVals(RDVFinalCleanupList &list) {
   LDBG() << "Starting cleanup of dead values...";
 
-  // 1. Operations
+  // 1. Blocks, We must remove the block arguments and successor operands before
```

Comment:
So this is a case of "delete stuff inside the operation before the operation itself"? Do we also guarantee that when erasing operations?

Reply:
> "delete stuff inside the operation before the operation itself"?
yes.
> "Do we also guarantee that when erasing operations?"
now we remove block argument  and successorOperands before delete ops, I think we have guaranteed it.